tests/integration/modifiers/did-intersect-test.js

```diff
@@ -1,87 +1,97 @@
 
-    this.intersectStub = sinon.stub();
+    this.observerManagerMock = this.owner.lookup('service:ember-scroll-modifiers@observer-manager');
+    this.enterStub = sinon.stub();
+    this.exitStub = sinon.stub();
   });
 
-  hooks.afterEach(function () {
-    window.IntersectionObserver = intersectionObserver;
-  });
+  test('modifier integrates with observer-manager and triggers correct callbacks when onEnter and onExit are provided', async function (assert) {
+    assert.expect(3);
 
-  test('modifier integrates with IntersectionObserver', async function (assert) {
-    await render(hbs`<div {{did-intersect this.intersectStub}}></div>`);
+    await render(hbs`<div {{did-intersect onEnter=this.enterStub onExit=this.exitStub}}></div>`);
 
-    assert.ok(intersectionCallback, 'IntersectionObserver received callback');
-    assert.ok(observeStub.calledOnce, 'observe was called');
-    assert.ok(intersectionOptions, 'options passed to IntersectionObserver');
-    assert.equal(typeof intersectionOptions, 'object', 'default options is an object');
-    assert.equal(Object.keys(intersectionOptions).length, 0, 'default options is an empty object');
+    assert.ok(this.observerManagerMock.observe.calledOnce, 'observerManager received observe call');
+    assert.ok(this.observerManagerMock.addEnterCallback.calledOnce, 'observerManager received enter callback');
+    assert.ok(this.observerManagerMock.addExitCallback.calledOnce, 'observerManager received exit callback');
+  });
+
+  test('modifier triggers addEnterCallback but not addExitCallback if only onEnter is provided', async function (assert) {
+    assert.expect(2);
 
-    let [element] = observeStub.args[0];
+    await render(hbs`<div {{did-intersect onEnter=this.enterStub}}></div>`);
 
-    assert.ok(element, 'element was passed to observe');
+    assert.ok(this.observerManagerMock.addEnterCallback.calledOnce, 'observerManager received enter callback');
+    assert.notOk(this.observerManagerMock.addExitCallback.calledOnce, 'observerManager did not receive exit callback');
   });
 
-  test('modifier triggers handler when IntersectionObserver fires callback', async function (assert) {
-    await render(hbs`<div {{did-intersect this.intersectStub}}></div>`);
+  test('modifier triggers addExitCallback but not addEnterCallback if only onExit is provided', async function (assert) {
+    assert.expect(2);
+
+    await render(hbs`<div {{did-intersect onExit=this.exitStub}}></div>`);
 
-    let fakeEntry = { target: {} };
-    let fakeObserver = { observe: {} };
+    assert.notOk(this.observerManagerMock.addEnterCallback.calledOnce, 'observerManager did not receive enter callback');
+    assert.ok(this.observerManagerMock.addExitCallback.calledOnce, 'observerManager received exit callback');
+  });
+
+  test('modifier throws assertion if neither onEnter or onExit is provided', async function (assert) {
+    assert.expect(3);
```

Comment:
Keep this `assert.expect`, this is the only one we need since one of the assertions is in the callback of setupOnerror